src/components/StaticWebsite.ts

```diff
@@ -77,7 +79,26 @@ export class StaticWebsite extends CdkConstruct implements Construct {
             comment: `Identity that represents CloudFront for the ${id} static website.`,
         });
 
+        /**
+         * CloudFront function that manipulates the HTTP responses to add security headers.
+         */
+        const code = `function handler(event) {
+    var response = event.response;
+    response.headers = Object.assign({}, {
+        'x-frame-options': {value: 'SAMEORIGIN'},
+        'x-content-type-options': {value: 'nosniff'},
+        'x-xss-protection': {value: '1; mode=block'},
+        'strict-transport-security': {value: 'max-age=63072000'},
+    }, response.headers);
+    return response;
+}`;
+        const edgeFunction = new cloudfront.Function(this, "EdgeFunction", {
+            functionName: `${this.provider.stackName}-${this.provider.region}-${id}-response`,
+            code: cloudfront.FunctionCode.fromInline(code),
+        });
+
         const distribution = new CloudFrontWebDistribution(this, "CDN", {
+            comment: `${provider.stackName} ${id} website CDN`,
```

Comment:
Just to help users figure things out in the console:

<img width="772" alt="image" src="https://user-images.githubusercontent.com/720328/120812823-a49f7780-c54d-11eb-88fa-112c96ea988a.png">